vignettes/TADAGeospatialFunctions.Rmd

```diff
@@ -0,0 +1,236 @@
+```
+
+## `TADA_GetATTAINS()`
+
+This function pulls in ATTAINS data from the EPA's ATTAINS Assessment
+Geospatial Service and links it to TADA-pulled Water Quality Portal
+observations. For the function to work properly, the input data frame must
+have - at a minimum - WQP observation coordinates in "LongitudeMeasure" and
+"LatitudeMeasure" columns (or "TADA.LongitudeMeasure" and
+"TADA.LatitudeMeasures" columns if the user chose to auto-clean their WQP
+results) and a "HorizontalCoordinateReferenceSystemDatumName" column.
+
+Users also have the option of loading the ATTAINS geospatial {sf} objects into
+their local environment with `return = TRUE`. The objects returned are named
+`ATTAINS_catchments`, `ATTAINS_lines`, `ATTAINS_points`, and
+`ATTAINS_polygons`.
+
+Regardless of the user's decision on returning the ATTAINS {sf} objects,
+`TADA_GetATTAINS()` always returns a dataframe containing the original TADA
+WQP dataset, plus new columns representing the ATTAINS assessment unit(s) that
+fall within the same NHDPlus HR catchment as them. This means that it is
+possible for a single TADA WQP observation to have multiple ATTAINS assessment
+units linked to it and subsequently more than one row of data. Such WQP
+observations can be identified using the `index` column (i.e., multiple rows
+with the same index value are the same observation).
+
+#### Using `TADA_GetATTAINS()`
+
+Using either our original `TADA_dataframe` *or* the geospatial version
+`TADA_spatial`, we can pull in the ATTAINS features that are within the same
+NHD HR catchment as our observations:
+
+```{r}
+TADA_with_ATTAINS <- TADA_GetATTAINS(data = TADA_dataframe, return = TRUE)
+
+TADA_with_ATTAINS <- TADA_GetATTAINS(data = TADA_spatial, return = TRUE)
+```
+
+This new `TADA_with_ATTAINS` object is identical to the original TADA Water
+Quality Portal dataframe, but now has additional columns associated with the
+ATTAINS assessment unit(s) that lie in the same NHD HR catchment as them
+(these columns are prefixed with "ATTAINS"). However, because our
+`TADA_with_ATTAINS` object contains more rows than the original TADA
+dataframe, we can deduce that some water quality observations fall within an
+NHD HR catchment that contains multiple ATTAINS assessment units.
+
+And, because we set `return = TRUE`, we now have all the raw ATTAINS features
+associated with these TADA Water Quality Portal observations saved in our
+Global Environment as `ATTAINS_catchments`, `ATTAINS_lines`, `ATTAINS_points`,
+and `ATTAINS_polygons` objects.
+
+Enter `?TADA_GetATTAINS` into the console to review another example of this
+function in use and additional information.
+
+## `TADA_ViewATTAINS()`
+
+This function visualizes the raw ATTAINS features that are linked to the TADA
+WQP observations. (Essentially, this maps the same data created from running
+`TADA_GetATTAINS()`.) For the function to work properly, the input data frame
+must be a WQP dataframe developed by `TADA_DataRetrieval()`.
```

Comment:
I can't remember if we've discussed this (and couldn't find a mention) but would it make more sense to have `TADA_ViewATTAINS()` directly use the output of `TADA_GetATTAINS()`? I'm not sure how important of a point this is to consider, but it confused me a little in the flow of the vignette because it felt like i was downloading data that would be used in the next step. So this could end up with redundant the user doing redundant work

Reply:
Are you suggesting that we combine the functionality of both? As in, we add an argument to TADA_GetATTAINS() - like `visualize = TRUE` - that creates the leaflet map if the user want to view it?

Reply:
I could add some code to ensure that if the user supplied the dataframe produced from TADA_GetATTAINS(), it wouldn't break the code. Would that be helpful?